File: csharp/Uhflib/UHFReader.cs
using System;
using System.IO.Ports;

namespace Uhflib
{
    public class UHFReader
    {
        private SerialPort serialPort;

        public UHFReader(string comPort, int baudRate)
        {
            serialPort = new SerialPort(comPort, baudRate)
            {
                Parity = Parity.None,
                DataBits = 8,
                StopBits = StopBits.One,
                Handshake = Handshake.None
            };
        }

        public void Connect()
        {
            serialPort.Open();
            Console.WriteLine("Connected to UHF Reader.");
        }

        public void Disconnect()
        {
            serialPort.Close();
            Console.WriteLine("Disconnected from UHF Reader.");
        }

        public string ReadTag()
        {
            return serialPort.ReadLine();
        }

        public void SetPowerLevel(int dBm)
        {
            serialPort.WriteLine($"SET_PWR {dBm}");
        }

        public void SetPingRate(int milliseconds)
        {
            serialPort.WriteLine($"SET_PING {milliseconds}");
        }
    }
}

File: js/uhflib/UHFReader.js
const SerialPort = require('serialport');
const Readline = require('@serialport/parser-readline');

class UHFReader {
    constructor(comPort, baudRate) {
        this.port = new SerialPort(comPort, { baudRate: baudRate });
        this.parser = this.port.pipe(new Readline({ delimiter: '\n' }));
    }

    connect() {
        this.port.on('open', () => {
            console.log('Connected to UHF Reader.');
        });
    }

    disconnect() {
        this.port.close(() => {
            console.log('Disconnected from UHF Reader.');
        });
    }

    readTag(callback) {
        this.parser.once('data', (data) => {
            callback(data.trim());
        });
    }

    setPowerLevel(dBm) {
        this.port.write(`SET_PWR ${dBm}\n`);
    }

    setPingRate(milliseconds) {
        this.port.write(`SET_PING ${milliseconds}\n`);
    }
}

module.exports = UHFReader;